File: samples/MiscellaneousSamples/app/src/com/pdftron/android/pdfnetsdksamples/MiscellaneousSamplesApplication.java
//---------------------------------------------------------------------------------------
// Copyright (c) 2001-2019 by PDFTron Systems Inc. All Rights Reserved.
// Consult legal.txt regarding legal and license information.
//---------------------------------------------------------------------------------------

package com.pdftron.android.pdfnetsdksamples;

import android.app.Application;
import android.content.Context;

import com.pdftron.android.pdfnetsdksamples.samples.AddImageTest;
import com.pdftron.android.pdfnetsdksamples.samples.AnnotationTest;
import com.pdftron.android.pdfnetsdksamples.samples.BookmarkTest;
import com.pdftron.android.pdfnetsdksamples.samples.ContentReplacerTest;
import com.pdftron.android.pdfnetsdksamples.samples.ConvertTest;
import com.pdftron.android.pdfnetsdksamples.samples.DigitalSignaturesTest;
import com.pdftron.android.pdfnetsdksamples.samples.ElementBuilderTest;
import com.pdftron.android.pdfnetsdksamples.samples.ElementEditTest;
import com.pdftron.android.pdfnetsdksamples.samples.ElementReaderAdvTest;
import com.pdftron.android.pdfnetsdksamples.samples.ElementReaderTest;
import com.pdftron.android.pdfnetsdksamples.samples.EncTest;
import com.pdftron.android.pdfnetsdksamples.samples.FDFTest;
import com.pdftron.android.pdfnetsdksamples.samples.ImageExtractTest;
import com.pdftron.android.pdfnetsdksamples.samples.ImpositionTest;
import com.pdftron.android.pdfnetsdksamples.samples.InteractiveFormsTest;
import com.pdftron.android.pdfnetsdksamples.samples.JBIG2Test;
import com.pdftron.android.pdfnetsdksamples.samples.LogicalStructureTest;
import com.pdftron.android.pdfnetsdksamples.samples.OfficeToPDFTest;
import com.pdftron.android.pdfnetsdksamples.samples.OptimizerTest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFATest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFDocMemoryTest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFDrawTest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFLayersTest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFPackageTest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFPageTest;
import com.pdftron.android.pdfnetsdksamples.samples.PDFRedactTest;
import com.pdftron.android.pdfnetsdksamples.samples.PageLabelsTest;
import com.pdftron.android.pdfnetsdksamples.samples.PatternTest;
import com.pdftron.android.pdfnetsdksamples.samples.RectTest;
import com.pdftron.android.pdfnetsdksamples.samples.SDFTest;
import com.pdftron.android.pdfnetsdksamples.samples.StamperTest;
import com.pdftron.android.pdfnetsdksamples.samples.TextExtractTest;
import com.pdftron.android.pdfnetsdksamples.samples.TextSearchTest;
import com.pdftron.android.pdfnetsdksamples.samples.U3DTest;
import com.pdftron.android.pdfnetsdksamples.samples.UndoRedoTest;
import com.pdftron.android.pdfnetsdksamples.samples.UnicodeWriteTest;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

//import com.pdftron.android.pdfnetsdksamples.samples.DocxConvertTest;

public class MiscellaneousSamplesApplication extends Application {

    private ArrayList<PDFNetSample> mListSamples = new ArrayList<PDFNetSample>();
    private static MiscellaneousSamplesApplication singleton;
    private Context m_context;

    @Override
    public void onCreate() {
        super.onCreate();
        singleton = this;
        mListSamples.add(new AddImageTest());
        mListSamples.add(new AnnotationTest());
        mListSamples.add(new BookmarkTest());
        mListSamples.add(new ContentReplacerTest());
        mListSamples.add(new ConvertTest());
        mListSamples.add(new DigitalSignaturesTest());
        mListSamples.add(new ElementBuilderTest());
        mListSamples.add(new ElementEditTest());
        mListSamples.add(new ElementReaderTest());
        mListSamples.add(new ElementReaderAdvTest());
        mListSamples.add(new EncTest());
        mListSamples.add(new FDFTest());
        mListSamples.add(new ImageExtractTest());
        mListSamples.add(new ImpositionTest());
        mListSamples.add(new InteractiveFormsTest());
        mListSamples.add(new JBIG2Test());
        mListSamples.add(new LogicalStructureTest());
        mListSamples.add(new OptimizerTest());
        mListSamples.add(new PageLabelsTest());
        mListSamples.add(new PatternTest());
        mListSamples.add(new PDFATest());
        mListSamples.add(new PDFDocMemoryTest());
        mListSamples.add(new PDFDrawTest());
        mListSamples.add(new PDFLayersTest());
        mListSamples.add(new PDFPackageTest());
        mListSamples.add(new PDFPageTest());
        mListSamples.add(new PDFRedactTest());
        mListSamples.add(new RectTest());
        mListSamples.add(new SDFTest());
        mListSamples.add(new StamperTest());
        mListSamples.add(new TextExtractTest());
        mListSamples.add(new TextSearchTest());
        mListSamples.add(new U3DTest());
        mListSamples.add(new UnicodeWriteTest());
        mListSamples.add(new OfficeToPDFTest(getApplicationContext()));
        mListSamples.add(new UndoRedoTest());

        Collections.sort(mListSamples, new Comparator<PDFNetSample>() {
            @Override
            public int compare(PDFNetSample o1, PDFNetSample o2) {
                return o1.getTitle().compareTo(o2.getTitle());
            }
        });

        m_context = getApplicationContext();
    }

    public List<PDFNetSample> getContent() {
        return this.mListSamples;
    }

    public static MiscellaneousSamplesApplication getInstance() {
        return singleton;
    }

    public Context getContext() {
        return m_context;
    }
}

File: samples/MiscellaneousSamplesKotlin/app/src/com/pdftron/android/pdfnetsdksamples/MiscellaneousSamplesApplication.kt
//---------------------------------------------------------------------------------------
// Copyright (c) 2001-2018 by PDFTron Systems Inc. All Rights Reserved.
// Consult legal.txt regarding legal and license information.
//---------------------------------------------------------------------------------------

package com.pdftron.android.pdfnetsdksamples

import android.app.Application
import android.content.Context
import com.pdftron.android.pdfnetsdksamples.samples.*
import java.util.*

class MiscellaneousSamplesApplication : Application() {

    private val mListSamples = ArrayList<PDFNetSample>()
    var context: Context? = null
        private set

    val content: List<PDFNetSample>
        get() = this.mListSamples

    override fun onCreate() {
        super.onCreate()
        instance = this
        mListSamples.add(AddImageTest())
        mListSamples.add(AnnotationTest())
        mListSamples.add(BookmarkTest())
        mListSamples.add(ContentReplacerTest())
        mListSamples.add(ConvertTest())
        mListSamples.add(DigitalSignaturesTest())
        mListSamples.add(ElementBuilderTest())
        mListSamples.add(ElementEditTest())
        mListSamples.add(ElementReaderTest())
        mListSamples.add(ElementReaderAdvTest())
        mListSamples.add(EncTest())
        mListSamples.add(FDFTest())
        mListSamples.add(ImageExtractTest())
        mListSamples.add(ImpositionTest())
        mListSamples.add(InteractiveFormsTest())
        mListSamples.add(JBIG2Test())
        mListSamples.add(LogicalStructureTest())
        mListSamples.add(OptimizerTest())
        mListSamples.add(PageLabelsTest())
        mListSamples.add(PatternTest())
        mListSamples.add(PDFATest())
        mListSamples.add(PDFDocMemoryTest())
        mListSamples.add(PDFDrawTest())
        mListSamples.add(PDFLayersTest())
        mListSamples.add(PDFPackageTest())
        mListSamples.add(PDFPageTest())
        mListSamples.add(PDFRedactTest())
        mListSamples.add(RectTest())
        mListSamples.add(SDFTest())
        mListSamples.add(StamperTest())
        mListSamples.add(TextExtractTest())
        mListSamples.add(TextSearchTest())
        mListSamples.add(U3DTest())
        mListSamples.add(UnicodeWriteTest())
        mListSamples.add(OfficeToPDFTest(applicationContext))
        mListSamples.add(UndoRedoTest())

        mListSamples.sortWith(Comparator { o1, o2 -> o1.title!!.compareTo(o2.title!!) })

        context = applicationContext
    }

    companion object {
        var instance: MiscellaneousSamplesApplication? = null
            private set
    }
}
